starboard/tools/testing/test_coverage.py

```diff
@@ -0,0 +1,168 @@
+  llvm_cov = os.path.join(toolchain_dir, 'llvm-cov')
+
+  exclude_opts = _get_exclude_opts()
+
+  test_binary_opts = [target_paths[0]] + \
+    ['-object=' + target for target in target_paths[1:]]
+  if report_type in ['html', 'both']:
+    logging.info('Creating html coverage report')
+    show_cmd_list = [
+        llvm_cov, 'show', *exclude_opts, '-instr-profile=' + merged_data_path,
+        '-format=html', '-output-dir=' + os.path.join(report_dir, 'html'),
+        *test_binary_opts
+    ]
+    subprocess.check_call(show_cmd_list, text=True)
+
+  if report_type in ['text', 'both']:
+    logging.info('Creating text coverage report')
+    report_cmd_list = [
+        llvm_cov, 'show', *exclude_opts, '-instr-profile=' + merged_data_path,
+        '-format=text', *test_binary_opts
+    ]
+    with open(os.path.join(report_dir, 'report.txt'), 'wb') as out:
+      subprocess.check_call(report_cmd_list, stdout=out, text=True)
+
+
+def _merge_coverage_data(targets, coverage_dir):
+  """Create a combined coverage file from the raw reports."""
+  toolchain_dir = build.GetClangBinPath(clang.GetClangSpecification())
+  llvm_profdata = os.path.join(toolchain_dir, 'llvm-profdata')
+
+  raw_reports = _get_raw_reports(targets, coverage_dir)
+  if not raw_reports:
+    raise RuntimeError('No coverage data found in \'coverage_dir\'')
+
+  logging.info('Merging coverage files')
+  merged_data_path = os.path.join(coverage_dir, 'report.profdata')
+  merge_cmd_list = [
+      llvm_profdata, 'merge', '-sparse=true', '-o', merged_data_path
+  ] + raw_reports
+  subprocess.check_call(merge_cmd_list, text=True)
+
+  return merged_data_path
+
+
+def create_report(binary_dir, test_binaries, coverage_dir, report_type='both'):
+  """Generate source code coverage reports."""
+  merged_data_path = _merge_coverage_data(test_binaries, coverage_dir)
+
+  # TODO(oxv): Make configurable?
+  report_dir = coverage_dir
+  target_paths = _get_test_binary_paths(binary_dir, test_binaries, coverage_dir)
+  _generate_reports(target_paths, merged_data_path, report_dir, report_type)
+
+
+if __name__ == '__main__':
+  SetupDefaultLoggingConfig()
+
+  def report_type_checker(param):
+    if param not in ['text', 'html', 'both']:
+      raise argparse.ArgumentTypeError('invalid value: ' + param)
+    return param
+
+  arg_parser = argparse.ArgumentParser()
+  arg_parser.add_argument(
+      '--binaries_dir',
+      required=True,
+      help='Directory where the test binaries are located.')
+  arg_parser.add_argument(
+      '--coverage_dir',
+      required=True,
+      help='Directory where coverage files are located and where the reports '
+      'will be generated.')
+  arg_parser.add_argument(
+      '--test_binaries',
+      nargs='+',
+      default=[],
+      help='List of test binaries that should be used for the coverage report. '
+      'If not passed all coverage files in --coverage_dir will be used.')
+  arg_parser.add_argument(
```

Comment:
Consider instead adding two flags where each removes a report type from the default set. I.e., `--no-html` and `--no-text`. I think that may be a bit more clear.

Reply:
Refactored. Instead of this I went with `<type>-only` parameters using a mutual exclusion group.